src/providers/net.rs

```diff
@@ -0,0 +1,104 @@
+// Hegemon - A modular system monitor
+// Copyright (C) 2018-2020  Philipp Emanuel Weidmann <pew@worldwidemann.com>
+// Copyright (C) 2020       Astro <astro@spaceboyz.net>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::time::Instant;
+
+use systemstat::{Platform, System};
+
+use crate::stream::{Stream, StreamProvider};
+
+pub struct BandwidthStreamProvider {}
+
+impl StreamProvider for BandwidthStreamProvider {
+    fn streams(&self) -> Vec<Box<dyn Stream>> {
+        let mut streams = Vec::new();
+        let platform = System::new();
+
+        if let Ok(networks) = platform.networks() {
+            for net in networks.values() {
+                let name = net.name.clone();
+                streams.push(Stream::new(
+                    format!("rx:{}", name),
```

Comment:
```suggestion
                    format!("{}Rx", name),
```

This more closely aligns with the existing streams. Same applies for the Tx stream of course. I'm exploring alternative naming schemes for streams and this might change in the future, but for now we need to remain consistent.